de-de/c-de.html.markdown

```diff
@@ -0,0 +1,880 @@
+    0x0F | 0xF0; // => 0xFF (Bitweises ODER)
+    0x04 ^ 0x0F; // => 0x0B (Bitweises XOR)
+    0x01 << 1; // => 0x02 (Bitweises  Linksverschiebung (left shift) (um 1))
+    0x02 >> 1; // => 0x01 (Bitweises Rechtsverschiebung (right shift) (um 1))
+
+    // Sei vorsichtig beim Shift mit vorzeichenbehafteten Integern
+    // folgende Ausdrücke sind nicht definiert:
+    // - Verschiebung in das Vorzeichenbit (int a = 1 << 31)
+    // - Linksshift einer negativen Zahl (int a = -1 << 2)
+    // - Shift um einen Offset, welcher >= die Breite des linken Ausdrucks ist.
+    // int a = 1 << 32; // undefiniertes Verhalten, wenn int 32-Bit ist.
+    
+    ////////////////////////////////////////////////
+    // Kontrollstrukturen
+    ////////////////////////////////////////////////
+    if (0) {
+        printf("Ich werde nie ausgeführt.");
+    }
+    else if (0){
+        printf("Ich werde auch nie ausgeführt.");
+    }
+    else {
+        printf("Ich gebe etwas aus.");
+    }
+
+    // While-Schleifen existieren auch
+    int ii = 0;
+    while (ii < 10){ // JEDER Wert unter zehn ist wahr
+        printf("%d, " ii++); //i++ inkrementiert ii NACHDEM der Wert gebraucht
+                             // wurde.
+    } // => gibt folgendes aus: "0, 1, 2, 3, 4, 5, 6, 7, 8, 9, "
+
+    printf("\n");
+
+    int kk = 0;
+    do {
+        printf("%d, ", kk);
+    } while(++kk < 10); //++kk inkrementiert kk BEVOR der Wert gebraucht wurde.
+    // => gibt folgendes aus: "0, 1, 2, 3, 4, 5, 6, 7, 8, 9, "
+
+    printf("\n");
+
+    // In C gibt es auch for-Schleifen
+    int jj; 
+    for (jj = 0; jj < 10; jj++){
+        printf("%d, ", jj);
+    } // => gibt folgendes aus: "0, 1, 2, 3, 4, 5, 6, 7, 8, 9, "
+    
+    printf("\n");
+
+    // **Merke**
+    // Schleifen und Funktionen müssen einen Rumpf haben. Wenn kein Rumpf gebraucht
+    // wird, kann folgendes gemacht werden:
+    int i; 
+    for (i = 0; i <= 5; i++){
+        ; // Semikolon wird als Rumpf behandelt (Null-Anweisung)
+    }
+    // Alternativ kann auch folgendes geschrieben werden:
+    for (i = 0; i <= 5; i++);
+
+    // Verzweigungen mit mehreren Möglichkeiten: `switch()`
+    switch (a){
+        case 0: // Labels müssen integrale *konstante* Ausdrücke sein (z.B. Enums)
+            printf("Hey, 'a' ist gleich 0!\n");
+            break; //Wenn du kein break einsetzt, so geht der Kontrollfluss
+                   // durch die Labels
+        case 1:
+            printf("Huh, 'a' ist gleich 1!\n");
+            break;
+            // Sei vorsichtig - wenn man das `break` vergisst, werden alle 
+            // Anweisungen ausgeführt bis das nächste `break` erscheint.
+        case 3:
+        case 4:
+            printf("Schau mal ... 'a' ist entweder 3 oder 4.\n");
+            break;
+        default:
+            // wenn der Ausdruck `a` auf kein Label zutrifft.
+            fputs("Fehler!\n", stderr);
+            exit(-1);
+            break;
+    }
+
+    // Verwendung von "goto" in C
```

Comment:
given the famous debate around goto, I don't think this section should be included in the tutorial

Reply:
Removed this section. You're right it should not be part in a tutorial.